### PR TITLE
:sparkles: Enable users to integrate external checks via blank-imports

### PR DESCRIPTION
**Pull request by pdxjohnny**
#### What kind of change does this PR introduce?

- [x] feat: expose checker registration
- [x] feat: enable checker go plugins
- [ ] feat: docs for checker go plugins
- [ ] feat: tests for checker go plugins
- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

- Checks must be added in tree within upstream scorecard repo

#### What is the new behavior (if this is a feature change)?**

- [ ] Plugins can be loaded by specifying a `SCORECARD_DYNAMIC_CHECKS`
  environment variable which leverages Golang plugins
- [ ] Docs have been added
  - [ ] 
- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

NONE

#### Special notes for your reviewer

**TODO**

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Check plugins using go plugins
```

### CODE REVIEW

File: checks/dynamic.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"plugin"
+	"strings"
+)
+
+const CheckDynamicPluginLocationEnvironmentVariable = "SCORECARD_DYNAMIC_CHECKS"
```
Comment (pdxjohnny) [inline checks/dynamic.go:25]: - [ ] Rename to `checks/plugins.go`

**Review by spencerschrock — CHANGES_REQUESTED** [review]
We are trying to avoid using plugins for the reasons specified in #3095

**Comment by spencerschrock** [comment]
cc @naveensrinivasan for possible coordination

**Comment by pdxjohnny** [comment]
Thanks @spencerschrock, so just to clarify focus of this PR would be on the check docs side of things?

**Comment by spencerschrock** [comment]
> Thanks @spencerschrock, so just to clarify focus of this PR would be on the check docs side of things?

There's still the aspect of the changes outlined in #3095, and the followup comment https://github.com/ossf/scorecard/issues/3095#issuecomment-1571046576.

**Comment by naveensrinivasan** [comment]
I apologize for the delayed response. I will be working on this. Thanks!

**Comment by pdxjohnny** [comment]
Should I close this PR then?

**Comment by spencerschrock** [comment]
> Should I close this PR then?

I'm not sure about Naveen's expected timeline, but I'll go ahead and close this for now. 
@naveensrinivasan can you update us if you dont end up pursuing this?

**Pull request closed without merging**